### PR TITLE
Add percentile PPL function

### PR DESCRIPTION
**Pull request by salyh**
### Description
This PR implements the percentile PPL stats function

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/482

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by salyh** [comment]
Should we also add PERCENTILE_APPROX? Would be a nobrainer imho.

**Comment by salyh** [comment]
Am I right that the percentileAggFunction https://github.com/eliatra/opensearch-spark/blob/percentile/ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4#L244 is something out of scope with regards to this PR?

**Comment by salyh** [comment]
Seems I missed https://github.com/opensearch-project/sql/blob/main/ppl/src/main/antlr/OpenSearchPPLParser.g4#L241C1-L244C5  althoug not sure what "compression" refers to?

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/utils/AggregatorTranslator.java [hunk 1]
```diff
@@ -35,6 +40,20 @@ static Expression aggregator(org.opensearch.sql.ast.expression.AggregateFunction
                 return new UnresolvedFunction(seq("COUNT"), seq(arg),false, empty(),false);
             case SUM:
                 return new UnresolvedFunction(seq("SUM"), seq(arg),false, empty(),false);
+            case PERCENTILE:
```
Comment (YANG-DB) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/utils/AggregatorTranslator.java:43]: @salyh please add logical plan tests and full query IT tests

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/utils/AggregatorTranslator.java (hunk 1 above)
Comment (kt-eliatra) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/utils/AggregatorTranslator.java:43]: done

**Review by YANG-DB — COMMENTED** [review]
Thanks
please add sample usage in the [README section](https://github.com/opensearch-project/opensearch-spark/blob/c733d3cb403e5e57c2c2c4625be7ec5ce8c8ff10/ppl-spark-integration/README.md?plain=1#L260)

**Comment by YANG-DB** [comment]
Regarding `percentileAggFunction` you are correct - this command is no longer in the [SQL PPL antlr](https://github.com/opensearch-project/sql/blob/main/ppl/src/main/antlr/OpenSearchPPLParser.g4)

**Comment by salyh** [comment]
> Thanks please add sample usage in the [README section](https://github.com/opensearch-project/opensearch-spark/blob/c733d3cb403e5e57c2c2c4625be7ec5ce8c8ff10/ppl-spark-integration/README.md?plain=1#L260)

can you pls first comment on:

- https://github.com/opensearch-project/opensearch-spark/pull/547#issuecomment-2278393281
- https://github.com/opensearch-project/opensearch-spark/pull/547#issuecomment-2278397379

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**